### PR TITLE
⬆️(core) upgrade ansible to 2.7

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

Now that we are running python 3, it's time to also update Arnold's core engine to Ansible 2.7. 

## Proposal

- [x] upgrade to Ansible 2.7.7
- [x] upgrade to openshift 0.8.4
- [x] renamed `openshift_raw` module to `k8s`
- [x] use `k8s_facts` module instead of the `k8s` lookup

### CODE REVIEW

**Comment by jmaupetit** [comment]
Nota bene: this is a draft work for curiosity. I just want to know what fails after switching to the new `k8s` module.

**Review by sampaccoud — APPROVED** [review]

File: .circleci/config.yml [hunk 1]
```diff
@@ -400,9 +400,55 @@ jobs:
...
+            bin/ci-test-service "richie" "Installation successful!" "/" "next"
+
+      - run:
+          name: Test "richie"'s next stack substitution (create + delete)
+          command: |
+            # Deploy the next stack a second time
+            bin/ci-ansible-playbook deploy.yml "richie"
+            # Test service re-deployed with the next route
+            bin/ci-test-service "richie" "Installation successful!" "/" "next"
+            # Test that only one stack has been deployed
```
Comment (lunika) [inline .circleci/config.yml]: I think this comment is not needed

File: bin/_config.sh [hunk 2]
```diff
@@ -42,10 +42,13 @@ function _set_openshift_env() {
 #
 # usage: _docker_run [OPTIONS] [COMMAND]
 #
-#   OPTIONS: docker run options
-#    --env-file: file to set environment variables (_e.g._ env.d/production)
-#
-#   COMMAND: command to be run in Arnold container (_e.g._ ansible-playbook deploy.yml)
+#   OPTIONS:
+#     -e    file to set environment variables (_e.g._ env.d/production)
+#     -m    mount sources as a volume
```
Comment (lunika) [inline bin/_config.sh]: You should provide an example. I have no idea how to use this option without viewing the code

File: bin/_config.sh [hunk 3]
```diff
@@ -42,10 +42,13 @@ function _set_openshift_env() {
...
+#     -t    allocate a pseudo-TTY while running command
```
Comment (lunika) [inline bin/_config.sh]: Same here, provide an example.

File: bin/_config.sh [hunk 4]
```diff
@@ -42,10 +42,13 @@ function _set_openshift_env() {
...
+#     -v    verbose mode
```
Comment (lunika) [inline bin/_config.sh]: Same here, provide an example. Should I use `-vv` or `-v=2` ?

Comment (lunika) [reply bin/_config.sh]: Sorry I thought this option was the ansible verbosity.

File: create_redirect.yml [hunk 5]
```diff
@@ -49,28 +49,21 @@
...
-              lookup(
-                'openshift',
-                api_version='v1',
-                namespace=project_name,
-                kind='Service',
-                resource_name='redirect-nginx'
-              )
-            ] | flatten
-          }}
+      k8s_facts:
```
Comment (lunika) [inline create_redirect.yml:52]: "coeur avec les doigts"

File: delete_project.yml [hunk 6]
```diff
@@ -15,9 +15,18 @@
 
     - import_tasks: tasks/set_vars.yml
 
+    - name: Check if project exists
```
Comment (lunika) [inline delete_project.yml:20]: Do you think this task is really needed ? It fails now if the project doesn't exist ?

Comment (jmaupetit) [reply delete_project.yml:20]: I should re-test it but yes: it failed during my first tests :cry:

**Comment by lunika** [comment]
I have an issue when I want to deploy marsha. pods, jobs end in error because the env variable `DJANGO_SENTRY_DSN` does not have a real dsn (a real url I think). This is error is not visible in `CI` env because the secret has `DJANGO_SENTRY_DSN` defined. I think the line 19 of file `apps/marsha/templates/app/secret.yml.j2` should be:

```
DJANGO_SENTRY_DSN: "{{ MARSHA_VAULT.DJANGO_SENTRY_DSN | default('') | b64encode }}"
```

File: .circleci/config.yml (hunk 1 above)
Comment (jmaupetit) [reply .circleci/config.yml]: Oops

File: delete_project.yml (hunk 6 above)
Comment (jmaupetit) [reply delete_project.yml:20]: You are right. Maybe there was a delay before deletion and I thought nothing happens. BTW this is now fixed.

**Comment by jmaupetit** [comment]
Can you declare an issue for Marsha's secret bug please?

I've addressed all of your comments in separated fixup commits. A quick review would be wonderful to make a new release :pray:

**Comment by lunika** [comment]
> Can you declare an issue for Marsha's secret bug please?

Yes I can.

**Review by lunika — APPROVED** [review]
rebase and merge :+1:

**Pull request merged**